feature_engine/selection/drop_psi_features.py

```diff
@@ -0,0 +1,566 @@
+
+        # Select features below the threshold
+        self.features_to_drop_ = self.psi_[
+            self.psi_.value >= self.threshold
+        ].index.to_list()
+
+        self.n_features_in_ = X.shape[1]
+
+        return self
+
+    def _compute_PSI(self, df_basis, df_meas, bucketer):
+        """
+        Compute the PSI by first bucketing (=binning) and then call the psi calculator
+        for each feature.
+
+        Parameters
+        ----------
+        df_basis : pandas dataframe of shape = [m_rows, n_features]
+            Serve as reference for the features distribution.
+
+        df_meas : pandas dataframe of shape = [p_rows, n_features]
+            Matrix for which the feature distributions will be compared to the basis.
+
+        bucketer : EqualFrequencyDiscretiser or EqualWidthDiscretiser.
+                Default = EqualFrequencyDiscretiser
+            Class used to bucket (bin) the features in order to approximate
+            the distribution.
+
+        Returns
+        -------
+        results_df: pandas dataframe.
+            Dataframe containing the PSI for each feature.
+        """
+        # Initialize a container for the results.
+        results = {}
+
+        # Compute the PSI for each feature excluding the column used for split.
+        for feature in self.variables_:
+            results[feature] = [
+                self._compute_feature_psi(df_basis[[feature]], df_meas[[feature]])
+            ]
+
+        # Transform the result container in a user friendly format.
+        results_df = pd.DataFrame.from_dict(results).T
+        results_df.columns = ["value"]
+
+        return results_df
+
+    def _compute_feature_psi(self, series_basis, series_meas):
+        """
+        Call the PSI calculator for two distributions
+
+        Parameters
+        ----------
+        series_basis : pandas series
+            Proxy for the reference distribution.
+
+        series_meas : pandas series
+            Proxy for the measurement distribution.
+
+        Returns
+        -------
+        psi_value: float.
+            PSI value.
+        """
+        # Perform the binning for all features.
+        basis_binned = (
+            self.bucketer.fit_transform(series_basis.dropna()).value_counts().fillna(0)
+        )
+
+        meas_binned = (
+            self.bucketer.transform(series_meas.dropna()).value_counts().fillna(0)
+        )
+
+        # Combine the two distributions by merging the buckets (bins)
+        binning = (
+            pd.DataFrame(basis_binned)
+            .merge(
+                pd.DataFrame(meas_binned),
+                right_index=True,
+                left_index=True,
+                how="outer",
+            )
+            .fillna(0)
+        )
+        binning.columns = ["basis", "meas"]
+
+        # Compute the PSI value.
+        psi_value = self._psi(binning.basis.to_numpy(), binning.meas.to_numpy())
```

Comment:
I would return the 2 distributions ready as input of the psi function and leave that calculation to the following function, so we have functions with clear names, and thus clear delimited functionality

Reply:
I restructured the code to make it simpler